skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -130,15 +156,165 @@ def pcoa(distance_matrix):
     eigvecs[:, num_positive:] = np.zeros(eigvecs[:, num_positive:].shape)
     eigvals[num_positive:] = np.zeros(eigvals[num_positive:].shape)
 
+    # Normalize eigenvectors to unit length
+    if normalize_eigenvectors:
+        eigvecs = np.apply_along_axis(lambda vec: vec / np.linalg.norm(vec),
+                                      axis=1, arr=eigvecs)
+
+    # In case eigh is used, eigh computes all eigenvectors and -values.
+    # So if to_dimension was specified, we manually need to ensure only the
+    # requested number of dimensions (number of eigenvectors and
+    # eigenvalues,
+    # respectively) are returned.
+    eigvecs = eigvecs[:, :to_dimension]
+    eigvals = eigvals[:to_dimension]
+
+    # Scale eigenvalues to have length = sqrt(eigenvalue). This
+    # works because np.linalg.eigh returns normalized
+    # eigenvectors. Each row contains the coordinates of the
+    # objects in the space of principal coordinates. Note that at
+    # least one eigenvalue is zero because only n-1 axes are
+    # needed to represent n points in an euclidean space.
     coordinates = eigvecs * np.sqrt(eigvals)
+
+    # Calculate the array of proportion of variance explained
     proportion_explained = eigvals / eigvals.sum()
 
-    axis_labels = ['PC%d' % i for i in range(1, eigvals.size + 1)]
+    axis_labels = list(["PC%d" % i for i in range(1, to_dimension + 1)])
     return OrdinationResults(
-        short_method_name='PCoA',
-        long_method_name='Principal Coordinate Analysis',
+        short_method_name="PCoA",
+        long_method_name=long_method_name,
         eigvals=pd.Series(eigvals, index=axis_labels),
-        samples=pd.DataFrame(coordinates, index=distance_matrix.ids,
+        samples=pd.DataFrame(coordinates, index=distance_matrix_obj.ids,
                              columns=axis_labels),
         proportion_explained=pd.Series(proportion_explained,
```

Comment:
Has the proportion explained with fsvd been resolved?

Reply:
not yet; pls see other comment thread below